src/d3a/d3a_core/simulation.py

```diff
@@ -361,6 +366,11 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
             self._update_and_send_results()
             if self.export_on_finish and not self.redis_connection.is_enabled():
                 self.export.data_to_csv(self.area, True if slot_no == 0 else False)
+                if self.area.current_market is not None:
```

Comment:
I would actually put this check into `raw_data_to_json` do also have this check on every place. Thanks!

Reply:
Again think from the perspective of performance here too, this function is only used in two places, in this place it is always needed to check `if self.area.current_market` since it loops from the start of the simulation where current market is None and in other place it is guaranteed that there would be `current_market ` so no need to have this check in second place, so indeed I  can put this check inside `raw_data_to_json ` and have an early return if `current_market is not None` but wouldnt that degrades the *performance* that you first have to jump inside the function and then do the same check, if the method `raw_data_to_json ` would have been used in a lot of places then I would definitely be in favour of it but here im skeptical to do so

Reply:
I am just realising, that you call `self.export.raw_data_to_json` twice ? why is this?
Addotionally, you need to check for `if self.area.current_market` every time you run `raw_data_to_json` anyway because you want to access a current_market member in there. So I have to be relentless here, and ask you to put the check into the function. Thanks!

Reply:
Adapted as per offline discussion, `raw_data_to_json `  is now being called only once now and `self.area.current_market` stays outside of the function next to other checks if raw data needs to be exported to disk or not